### ISSUE TITLE
Merge Ipython and metakernel magics

### ISSUE THREAD

**Issue by joequant** [Calysto/metakernel#206]
Right now Ipython and metakernel have separate standards for setting up magics, and this leads to maintainability issues.  It would be nice if there were a way of writing one set of code that would work for both.

**Comment by dsblank** [comment]
That is true, but I think unavoidable in most circumstances. Metakernel's magics are designed to be language independent, whereas IPython's are baked into a specific language (Python). Having said that, there may be more we can do to fix individual differences. Is there something specific that you are thinking about?

**Comment by joequant** [comment]
The problem is is that the language processing for the magics tends to be the same, and by having two different files, we are running into code forks where fixes for one code base aren't making it into the other.

In the worst case, we can have metakernel and Ipython have two different interface codes that call the same language processing module. Will take a look at what can be done.

**Comment by blink1073** [comment]
For reference, here is how [`xeus-python`](https://github.com/jupyter-xeus/xeus-python/blob/e8d51d331c3453fc19e5492c4e3af8249f9f11a6/src/xinteractiveshell.cpp#L21) uses the IPython magics directly.  However, they are also a Python kernel, as Doug mentioned.
